src/thunderbots/software/ai/hl/stp/tactic/goalie_tactic.cpp

```diff
@@ -49,14 +52,67 @@ double GoalieTactic::calculateRobotCost(const Robot &robot, const World &world)
     }
     else
     {
-        return 1000000;
+        return std::numeric_limits<int>::max() - 10;
```

Comment:
I'm not  sure I think that may be something @MathewMacDougall added, summoning him here

Reply:
Ah I did this to try prevent overflow during optimization with the hungarian alrogithm. Not suuuuper great, but should be removed once we have "proper" goalie assignment, which will be changed with some Play cleanup later.

Reply:
Sounds like we should create an issue# for this so we don't forget it.

Shouldn't using the std::numeric_limits prevent overflow?

Reply:
Created this ticket: https://github.com/UBC-Thunderbots/Software/issues/745

I understood `numeric_limits` was the largest number representable by the system (hardware). Don't think it handles overflow, because so many different operations could cause overflow later on in different ways.

Reply:
If that's the case, this doesn't look very safe with the -10 then

Reply:
Added TODO and ticket near line